gateway/gateway_test.go

```diff
@@ -376,7 +376,7 @@ var _ = Describe("Gateway", func() {
 
 			return validDataWithProperty
 		}
-		verifyEndpoing := func(endpoints []string, method string) {
+		verifyEndpoint := func(endpoints []string, method string) {
 			client := &http.Client{}
 			baseURL := "http://localhost:8080"
```

Comment:
We shouldn't use a static port for our unit tests. When running the test locally or in parallel this port might already be used by a different process or other tests. 

[freeport](github.com/rudderlabs/rudder-go-kit/testhelper) or [httptest.Server](https://pkg.go.dev/net/http/httptest) should be used instead.